### PR TITLE
Remove many GeometryInfo usages from examples

### PR DESCRIPTION
**Pull request by peterrum**
... from 63 to 21.

`GeometryInfo` is still used for following purposes:
- get the maximum number of faces/vertices to be able to create arrays with fixed sizes
- to get the face normals via `unit_normal_direction`
- to get the number of children via `max_children_per_cell` and `max_children_per_face`

### CODE REVIEW

File: examples/step-18/step-18.cc [hunk 1]
```diff
@@ -1482,7 +1482,7 @@ namespace Step18
   // points, see @ref GlossSupport "support points"). For such a case, one
   // could construct a custom quadrature rule using
   // FiniteElement::get_unit_support_points(). The first
-  // <code>GeometryInfo@<dim@>::%vertices_per_cell*fe.dofs_per_vertex</code>
+  // <code>cell-@>::%vertices_per_cell*fe.dofs_per_vertex</code>
```
Comment (drwells) [inline examples/step-18/step-18.cc:1485]:
```suggestion
  // <code>cell-&gt;n_vertices()*fe.dofs_per_vertex</code>
```

Comment (peterrum) [reply examples/step-18/step-18.cc]: I have applied the suggestion!

**Comment by luca-heltai** [comment]
If no-one objects, I'll merge tonight.

**Comment by bangerth** [comment]
Part of #11879.

**Pull request merged**